### PR TITLE
aws lb target group load balancing algorithm

### PR DESCRIPTION
**Pull request by wolviecb**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

The `load_balancing_algorithm` was added by AWS as a new ALB option as of Nov/25, please see [here](https://aws.amazon.com/about-aws/whats-new/2019/11/application-load-balancer-now-supports-least-outstanding-requests-algorithm-for-load-balancing-requests/).
I'm not very sure about the tests, might have tweaked too much or too little.
I've tested this and it looks ok :)

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
adds the argument `load_balancing_algorithm` to `aws_lb_target_group` resource and data sources. 
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSLBTargetGroup_least_outstanding_requests'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -count 1 -parallel 20 -run=TestAccAWSLBTargetGroup_least_outstanding_requests -timeout 120m
?       github.com/terraform-providers/terraform-provider-aws   [no test files]
=== RUN   TestAccAWSLBTargetGroup_least_outstanding_requests
=== PAUSE TestAccAWSLBTargetGroup_least_outstanding_requests
=== CONT  TestAccAWSLBTargetGroup_least_outstanding_requests
--- PASS: TestAccAWSLBTargetGroup_least_outstanding_requests (18.62s)
PASS
ok      github.com/terraform-providers/terraform-provider-aws/aws       20.575s
testing: warning: no tests to run
PASS
ok      github.com/terraform-providers/terraform-provider-aws/aws/internal/flatmap      0.076s [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags 0.196s [no tests to run]
```

### CODE REVIEW

File: website/docs/r/lb_target_group.html.markdown [hunk 1]
```diff
@@ -68,6 +68,7 @@ The following arguments are supported:
 * `slow_start` - (Optional) The amount time for targets to warm up before the load balancer sends them a full share of requests. The range is 30-900 seconds or 0 to disable. The default value is 0 seconds.
 * `lambda_multi_value_headers_enabled` - (Optional) Boolean whether the request and response headers exchanged between the load balancer and the Lambda function include arrays of values or strings. Only applies when `target_type` is `lambda`.
 * `proxy_protocol_v2` - (Optional) Boolean to enable / disable support for proxy protocol v2 on Network Load Balancers. See [doc](https://docs.aws.amazon.com/elasticloadbalancing/latest/network/load-balancer-target-groups.html#proxy-protocol) for more information.
+* `load_balancing_algorithm` - (Optional) The load balancing algorithm determines how the load balancer selects targets when routing requests. The value is `round_robin` or `least_outstanding_requests`. The default is `round_robin`. See [doc](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-target-groups.html#modify-routing-algorithm)
```
Comment (jonathanbeber) [inline website/docs/r/lb_target_group.html.markdown]:
```suggestion
* `load_balancing_algorithm` - (Optional) The load balancing algorithm determines how the load balancer selects targets when routing requests. The value is `round_robin` or `least_outstanding_requests`. The default is `round_robin`. When `least_outstanding_requests`, It cannot be enabled in conjunction with `slow_start`. See [doc](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-target-groups.html#modify-routing-algorithm) for more information.
```

File: website/docs/r/lb_target_group.html.markdown [hunk 2]
```diff
@@ -68,6 +68,7 @@ The following arguments are supported:
 * `slow_start` - (Optional) The amount time for targets to warm up before the load balancer sends them a full share of requests. The range is 30-900 seconds or 0 to disable. The default value is 0 seconds.
```
Comment (jonathanbeber) [inline website/docs/r/lb_target_group.html.markdown]:
```suggestion
 * `slow_start` - (Optional) The amount time for targets to warm up before the load balancer sends them a full share of requests. The range is 30-900 seconds or 0 to disable. The default value is 0 seconds. It cannot be enabled when `load_balance_algorithm` is `least_outstanding_requests`.
```

**Comment by wolviecb** [comment]
I've updated the PR to fix a merge conflict and also decide to just add the specific test function instead of mess around with other stuff on the _test file.

**Comment by bflad** [comment]
Hi @wolviecb 👋 Thank you for submitting this. This contribution appears to be very similar or duplicate to the earlier https://github.com/terraform-providers/terraform-provider-aws/pull/11141, so to consolidate efforts and discussions, I'm going to close this in preference of the earlier one.

**Pull request closed without merging**